mobile/library/common/jni/jni_interface.cc

```diff
@@ -627,16 +628,19 @@ jvm_http_filter_on_resume(const char* method, envoy_headers* headers, envoy_data
     headers_length = (jlong)headers->length;
     passHeaders("passHeader", *headers, j_context);
   }
-  jbyteArray j_in_data = nullptr;
+  Envoy::JNI::LocalRefUniquePtr<jbyteArray> j_in_data = Envoy::JNI::LocalRefUniquePtr<jbyteArray>(
+      nullptr, Envoy::JNI::LocalRefDeleter(jni_helper.getEnv()));
   if (data) {
-    j_in_data = Envoy::JNI::native_data_to_array(jni_helper, *data);
+    auto x = Envoy::JNI::native_data_to_array(jni_helper, *data);
+    j_in_data = std::move(x);
```

Comment:
curious, why do we need to do these two lines instead of just:
```j_in_data = Envoy::JNI::native_data_to_array(jni_helper, *data);```

Reply:
It should be one line like what you suggested. I was playing with the move semantics but forgot to remove that. Thanks for spotting it.